server/app/actions/RouteExtractor.java

```diff
@@ -0,0 +1,100 @@
+package actions;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * RouteExtractor interrogates the current request URI path and route pattern allowing for a simple
+ * way to get path parameters within a custom Action
+ *
+ * <p>The Play Framework does not currently expose URI path parameters within custom Actions. This
+ * is a simple workaround originally provided on the related Github Issue on the Play Framework. <a
+ * href="https://github.com/playframework/playframework/issues/2983#issuecomment-456375489">...</a>.
+ *
+ * <p>This is modified from the original. Created by Akinniranye James Ayodele on 8/12/16 12:56 PM.
+ * Project Andromeda Version 0.1
+ */
+public final class RouteExtractor {
+  private final String routePattern;
+  private final String path;
+  private final ImmutableMap<String, String> routeParameters;
+
+  // Example: something like /$id<[^/]+>
```

Comment:
suggestion (non-blocking):  I may not be the only one that doesn't know what this regex does off the top of my head.  Would you mind giving a one-line summary of what it does?